src/controllers/assistant.js

```diff
@@ -13,34 +13,137 @@
  * @returns {object} Import assistant controller.
  * @constructor
  */
-function AssistantController() {
+function AssistantController(context) {
   const HEADER_ERROR = 'x-error';
+  const { auth, log } = context;
 
   function createErrorResponse(error) {
     return createResponse({}, error.status, {
       [HEADER_ERROR]: error.message,
     });
   }
 
+  function isBase64UrlImage(base64String) {
+    return base64String.startsWith('data:image/') && base64String.endsWith('=') && base64String.includes('base64');
+  }
+
+  function validateRequestData(data) {
+    const {
+      command, prompt, imageUrl,
+    } = data;
+
+    // Validate 'command'
+    if (!command) {
+      throw new ErrorWithStatusCode('Invalid request: command is required.', STATUS.BAD_REQUEST);
+    }
+    const currentCommandConfig = commandConfig[command];
+    if (!currentCommandConfig) {
+      throw new ErrorWithStatusCode(`Invalid request: command not implemented: ${command}`, STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('prompt') && !prompt) {
+      throw new ErrorWithStatusCode('Invalid request: prompt is required.', STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('imageUrl')) {
+      if (!imageUrl) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is required.', STATUS.BAD_REQUEST);
+      }
+      // Only base64 images for now.
+      if (!isBase64UrlImage(imageUrl)) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is not a base64 encoded image.', STATUS.BAD_REQUEST);
+      }
+    }
+  }
+
+  function parseRequestContext(requestContext) {
+    if (!requestContext || !isObject(requestContext)) {
+      throw new ErrorWithStatusCode('Invalid request: missing request context.', STATUS.BAD_REQUEST);
+    }
+    if (!requestContext.data || !requestContext.attributes) {
+      throw new ErrorWithStatusCode('Invalid request: invalid request context format.', STATUS.BAD_REQUEST);
+    }
+
+    let assistantConfiguration;
+    if (!context.env.ASSISTANT_CONFIGURATION) {
+      throw new ErrorWithStatusCode('The Assistant Configuration is not defined.', STATUS.SYS_ERROR);
+    }
+    try {
+      assistantConfiguration = JSON.parse(context.env.ASSISTANT_CONFIGURATION);
+    } catch (error) {
+      throw new ErrorWithStatusCode(`Could not parse the Assistant Configuration: ${error.message}`, STATUS.SYS_ERROR);
+    }
+
+    const { data: { command, prompt, options }, attributes } = requestContext;
+    const { imageUrl, ...otherOptions } = options || {};
+    const { authInfo: { profile } } = attributes;
+    const requestData = {
+      env: {
+        ...context.env,
+        ...assistantConfiguration,
+        FIREFALL_API_KEY: assistantConfiguration.IMS_CLIENT_ID,
+      },
+      command,
+      prompt,
+      imageUrl,
+      llmModel: commandConfig[command]?.llmModel,
+      responseFormat: commandConfig[command]?.responseFormat,
```

Comment:
Do we need to validate the command exists first or is the pattern to try to create the request data first and then validate afterwards that the command config actually exists? 

Also, I wonder if it would be worth destructuring out the values you want like you've done for other objects.

eg.
```
const { llmModel, responseFormat } = commandConfig[command] || {};
```

Reply:
If the command doesn't exist, then the _llmModel_ and _responseFormat_ will be undefined.  The validation happens on `requestData` (not the original request values) soon afterwards and the request will be rejected if _command_ doesn't exist.

Destructuring... sure. 👍

Reply:
I actually created an object, within each `commandConfig` that is just 'spread' in the requestData, to be processed later.  Seemed more contained and easier to add things later.